src/NServiceBus.AwsLambda.SQS/AwsLambdaEndpoint.cs

```diff
@@ -361,13 +363,22 @@ static bool IsMessageExpired(SQSEvent.SQSMessage receivedMessage, Dictionary<str
             return true;
         }
 
-        async Task ProcessMessageWithInMemoryRetries(Dictionary<string, string> headers, string nativeMessageId, ReadOnlyMemory<byte> body, ILambdaContext lambdaContext, CancellationToken token)
+        async Task ProcessMessageWithInMemoryRetries(Dictionary<string, string> headers, string nativeMessageId, ReadOnlyMemory<byte> body, Message nativeMessage, ILambdaContext lambdaContext, CancellationToken token)
         {
             var immediateProcessingAttempts = 0;
             var errorHandled = false;
 
             while (!errorHandled)
             {
+                // set the native message on the context for advanced usage scenario's
+                var context = new ContextBag();
+                context.Set(nativeMessage);
+
+                // We add it to the transport transaction to make it available in dispatching scenario's so we copy over message attributes when moving messages to the error/audit queue
+                var transportTransaction = new TransportTransaction();
+                transportTransaction.Set(nativeMessage);
```

Comment:
Should we also make the SQS Event available here since I guess that would be more the expectation for someone using lambda? I still like though that we also make the SQS Message available since that gives a good value proposition because the SQS code of a regular endpoint is compatible.

Reply:
I think it does make sense.  The downside is that the transport wouldn't know about it.  So if any mutations were to be made to it either with behaviors or in the handler, the changes would be lost in the error queue cases, etc.  I don't know if that is a concern or not though.  I can't really think of a reason one would want to mutate these objects since they're coming from SQS and should probably be considered immutable.